code/modules/mob/living/simple_animal/constructs.dm

```diff
@@ -272,3 +352,80 @@
 				return
 			cultist.loc = usr.loc
 			usr.visible_message("/red [cultist] appears in a flash of red light as [usr] glows with power")*/
+
+
+///ui stuff
+
+/mob/living/simple_animal/hostile/construct/armoured/Life()
+
+	..()
+	if(healths)
+		switch(health)
+			if(250 to INFINITY)		healths.icon_state = "juggernaut_health0"
+			if(208 to 249)			healths.icon_state = "juggernaut_health1"
+			if(167 to 207)			healths.icon_state = "juggernaut_health2"
+			if(125 to 166)			healths.icon_state = "juggernaut_health3"
+			if(84 to 124)			healths.icon_state = "juggernaut_health4"
+			if(42 to 83)			healths.icon_state = "juggernaut_health5"
+			if(1 to 41)				healths.icon_state = "juggernaut_health6"
+			else					healths.icon_state = "juggernaut_health7"
+
+
+/mob/living/simple_animal/hostile/construct/behemoth/Life()

```

Comment:
no no no no no no no
there is a proc exactly for this
`/mob/living/proc/handle_hud_icons_health()`